File: 20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    let count = 0;
    for (let i = 0; i < s.length; i++) {
        if (s[i] === "(" || s[i] === "[" || s[i] === "{") {
            count++;
            stack.push(s[i]);
        } else {
            const test = stack.pop();
            if (s[i] === ")" && test !== "(") return false;
            if (s[i] === "}" && test !== "{") return false;
            if (s[i] === "]" && test !== "[") return false;
            count--;
        }
    }
    return !count;
};

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        map = {"(": ")", "[": "]", "{": "}" }
        for i in range(len(s)):
            if s[i] in map:
                stack.append(s[i])
            else:
                if not stack or map[stack.pop()] != s[i]:
                    return False
        return not stack